### PR TITLE
NONPR-1659 Secure access to PDF Generation service via X-API-Key

### PR DESCRIPTION
**Pull request by greg-gajda**

### CODE REVIEW

File: src/it/resources/application.conf [hunk 1]
```diff
@@ -0,0 +1,20 @@
+generate-pdf {
+  auditing-enabled = false
+
+  templates = [
+    {
+      template-id = "trusts-5mld-0-6-0"
+      json-schema = "API#1584_Response_Schema-v0.6.0.json"
+      pdf-template = ""
```
Comment (craigtd) [inline src/it/resources/application.conf:8]: What does `pdf-template` refer to if we already have `template-id`?

File: src/it/scala/uk/gov/hmrc/nonrep/pdfs/server/ServiceIntSpec.scala [hunk 2]
```diff
@@ -74,23 +77,47 @@ class ServiceIntSpec extends AnyWordSpec with Matchers with ScalatestRouteTest w
...
+      val request = Post(s"$hostUrl/$service/template/unknown/signed-pdf").
+        withEntity(HttpEntity(sampleRequest_0_6_0)).
+        withHeaders(RawHeader("X-API-Key", apiKey))
       val responseFuture: Future[HttpResponse] = Http().singleRequest(request)
       whenReady(responseFuture) { res =>
         res.status shouldBe StatusCodes.NotFound
       }
     }
 
     "accept and respond when requested valid template" in {
```
Comment (craigtd) [inline src/it/scala/uk/gov/hmrc/nonrep/pdfs/server/ServiceIntSpec.scala:89]: perhaps rephrase to "accept and respond when valid template requested"

**Review by opetch — APPROVED** [review]
lgtm

File: src/main/resources/application.conf [hunk 3]
```diff
@@ -0,0 +1,38 @@
+generate-pdf {
+  auditing-enabled = false
+
+  templates = [
+    {
+      template-id = "trusts-5mld-0-6-0"
+      json-schema = "API#1584_Response_Schema-v0.6.0.json"
+      pdf-template = ""
```
Comment (craigtd) [inline src/main/resources/application.conf:8]: as per earlier comment

File: src/test/resources/application.conf [hunk 4]
```diff
@@ -0,0 +1,20 @@
+generate-pdf {
+  auditing-enabled = false
+
+  templates = [
+    {
+      template-id = "trusts-5mld-0-6-0"
+      json-schema = "API#1584_Response_Schema-v0.6.0.json"
+      pdf-template = ""
```
Comment (craigtd) [inline src/test/resources/application.conf:8]: as with previous comment

**Review by craigtd — APPROVED** [review]
Looks fine but with one query (repeated)

File: src/it/resources/application.conf (hunk 1 above)
Comment (greg-gajda) [reply src/it/resources/application.conf:8]: https://github.com/hmrc/nonrep-generate-pdf/blob/master/src/main/scala/uk/gov/hmrc/nonrep/pdfs/server/ServiceConfig.scala#L25

File: src/main/resources/application.conf (hunk 3 above)
Comment (greg-gajda) [reply src/main/resources/application.conf:8]: https://github.com/hmrc/nonrep-generate-pdf/blob/master/src/main/scala/uk/gov/hmrc/nonrep/pdfs/server/ServiceConfig.scala#L25

File: src/test/resources/application.conf (hunk 4 above)
Comment (greg-gajda) [reply src/test/resources/application.conf:8]: https://github.com/hmrc/nonrep-generate-pdf/blob/master/src/main/scala/uk/gov/hmrc/nonrep/pdfs/server/ServiceConfig.scala#L25

File: src/it/resources/application.conf (hunk 1 above)
Comment (craigtd) [reply src/it/resources/application.conf:8]: of course

**Pull request merged**